File: DaggerScope/app/src/main/java/com/example/daggerscope/modules/AppModule.java
package com.example.daggerscope.modules;

import android.util.Log;
import com.example.daggerscope.entity.Person;
import com.example.daggerscope.scopes.AppScope;
import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    @AppScope
    @Provides
    Person providePerson() {
        Log.e(">>>", "provide Person");
        return new Person(18, "Bob");
    }

}

File: AndroidArchitectureComponentsDaggerSample/app/src/main/java/io/github/hurshi/androidarchitecturecomponentsdaggersample/modules/AppModule.kt
package io.github.hurshi.androidarchitecturecomponentsdaggersample.modules

import dagger.Module
import dagger.Provides
import io.github.hurshi.androidarchitecturecomponentsdaggersample.entity.DaoSessionSimulate
import io.github.hurshi.androidarchitecturecomponentsdaggersample.scopes.AppScope

@Module
class AppModule {
    @AppScope
    @Provides
    internal fun providerDao(): DaoSessionSimulate {
        return DaoSessionSimulate("Hello, I am a DB")
    }
}